doc/R-vignette/igraph_extended_tutorial_R_vignette_english.Rmd

```diff
@@ -0,0 +1,682 @@
+summary(g)
+```
+
+Uses `make_tree` to generate a regular tree graph with 127 vertices, each vertex having two children (and one parent, of course). No matter how many times you call `make_tree`, the generated graph will always be the same if you use the same parameters:
+
+```{r}
+graph2 <- make_tree(127, 2, mode = "undirected")
+```
+
+```{r echo = TRUE}
+identical_graphs(graph1,graph2)
+```
+
+*Stochastic* generators produce a different graph each time, e.g. `sample_grg`:
+
+```{r echo = TRUE}
+graph1 <- sample_grg(100, 0.2)
+summary(graph1)
+```
+
+This generates a geometric random graph: *n* points are chosen randomly and uniformly inside the unit square and pairs of points closer to each other than a predefined distance *d* are connected by an edge. If you generate GRGs with the same parameters, they will be different:
+
+```{r echo = TRUE}
+graph2 <- sample_grg(100, 0.2)
+identical_graphs(graph1,graph2)
+```
+
+A slightly looser way to check if the graphs are equivalent is via `isomorphic()`. Two graphs are said to be isomorphic if they have the same number of components (vertices and edges) and maintain a one-to-one correspondence between vertices and edges, i.e., they are connected in the same way.
+
+```{r echo = TRUE}
+isomorphic(graph1,graph2)
+```
+
+Checking for isomorphism can take a while for large graphs (in this case, the answer can quickly be given by checking the degree sequence of the two graphs). When using `identical_graph`, the two graphs must have the same list of vertices and edges, in exactly the same order, with same directedness, and the two graphs must also have identical graph, vertex and edge attributes.
+
+## Setting and retrieving attributes
+
+In addition to IDs, vertex and edges can have *attributes* such as a name, coordinates for plotting, metadata, and weights. The graph itself can have such attributes too (e.g. a name, which will show in `summary`). In a sense, every `graph`, vertex and edge can be used as a R variable to store and retrieve these attributes.
+
+To demonstrate the use of attributes, let us create a simple social network:
+
+```{r}
+g <- make_graph(~ Alice-Bob:Claire:Frank, Claire-Alice:Dennis:Frank:Esther, George-Dennis:Frank, Dennis-Esther)
+```
+
+Each vertex represents a person, so we want to store ages, genders and type of connection. The $ operator is a shortcut to get and and set graph attributes. It is shorter and just as readable as `graph_attr` and `set_graph_attr`.
+
+```{r echo = TRUE}
+V(g)$age <- c(25, 31, 18, 23, 47, 22, 50) 
+V(g)$gender <- c("f", "m", "f", "m", "m", "f", "m")
+E(g)$is_formal <- c(FALSE, FALSE, TRUE, TRUE, TRUE, FALSE, TRUE, FALSE, FALSE)
+summary(g)
+```
+
+`V` and `E` are the standard way to obtain a sequence of all vertices and edges, respectively. This assigns an attribute to *all* vertices/edges at once. Another way to generate our social network is with the use of `set_vertex_attr` and `set_edge_attr` and the operator %>%:
```

Comment:
@adriandiazlab Let's add a short sentence here, in brackets: "(`is_formal` indicates if a connection between two people is formal or informal, i.e. colleagues or friends.)"